react/reacap-theme-creator-app/src/components/AllThemes.js

```diff
@@ -1,13 +1,22 @@
-import { themes } from "../db";
-import Theme from "./Theme";
 import "./AllThemes.css";
+import { useState } from "react";
+import SingleTheme from "./SingleTheme.js";
+import ThemePreview from "./ThemePreview.js";
+import { IconChevronDown, IconChevronUp } from "@tabler/icons-react";
+
+export default function AllThemes({ theme }) {
+  const [detailView, checkDetailState] = useState(false);
+
+  function toggleView() {
+    checkDetailState(!detailView);
+  }
 
-export default function AllThemes() {
   return (
-    <>
-      {themes.map((theme) => {
-        return <Theme key={theme.id} theme={theme} />;
-      })}
-    </>
+    <article className="theme-overview">
+      <button className="view-toggle" onClick={toggleView}>
+        {detailView ? <IconChevronUp /> : <IconChevronDown />}
+      </button>
```

Comment:
you could also render your "<h2>{theme.name}</h2>" inside your button and delete it in your ThemePreview.js and SingleTheme.js.

Reply:
thank you! I applied your suggestions for changes.